### ISSUE TITLE
Navigation: Tabbing through menu causes upward scroll on page

### ISSUE THREAD

**Issue by ryelle** [WordPress/wporg-mu-plugins#147]
Reported in slack by @joedolson

> I'm not sure where to report this issue, as I can't really tell where it's coming from, but navigating the new header menu while in trac seems to cause some strange positional shifts in the page content. Gif coming to show it. I've only been able to reproduce this on trac, so far.
>
> ![keyboard-shift](https://user-images.githubusercontent.com/541093/152585250-016613e9-432e-450d-8c2e-4734b8a603c8.gif)
> The gif shows me tabbing through the main menu, and the trac content shifting with each keypress.

I was able to reproduce the same issue on make.wordpress.org, in firefox & safari. I imagine any long page will show the issue.

To reproduce:

1. Scroll partway down a page
2. Tab through the header & nav menu
3. Watch as the page content scrolls up slightly with each tab

**Comment by dd32** [comment]
I was expecting this to be caused by some JS attempting to ensure that anchors are on the screen, but that's not the case, it's reproducible with JS disabled.

This appears to be caused by:
```
html {
    height: unset;
    scroll-padding-top: 60px;
    scroll-padding-top: calc(var(--wp-global-header-height, 0px) + var(--wp-admin--admin-bar--height, 0px));
}
```

Interestingly, disabling the both of above rules and leaving the admin-bar version of it, doesn't cause jumping:
```
html {
    --wp-admin--admin-bar--height: 32px;
    scroll-padding-top: var(--wp-admin--admin-bar--height);
}
```

**Comment by joedolson** [comment]
It also goes away if you change the display of `header.global-header` from `flex` to `block` or if you change the positioning of `.wp-block-group.global-header` from `sticky` to `fixed`. Those also destroy the layout, but might help with understanding what's going on.

**Comment by ryelle** [comment]
I was able to replicate this same behavior [in plain HTML/CSS](https://codepen.io/ryelle/pen/wvPgrKY?editors=0100), so I can at least say the issue is the scroll-padding-top rule. It _seems_ to consider anything in that section of the page as "off the page", so when tabbing through, it's trying to bring it back "into view". 

I don't want to get rid of the `scroll-padding-top` rule, since that prevents the header from overlapping content when using relative links.

> Interestingly, disabling the both of above rules and leaving the admin-bar version of it, doesn't cause jumping:

Does it jump if you nav through the admin bar?

**Comment by dd32** [comment]
> Does it jump if you nav through the admin bar?

**Comment by dd32** [comment]
> Does it jump if you nav through the admin bar?

No, I can't reproduce any jumping with the admin bar, only with the header. I assumed it's the difference between `position: sticky` (header) and `position: fixed` (admin bar).

I had thought it might have something to do with `html { margin-top: 32px !important; }` as used by the admin bar, but I can't resolve the jumping with playing with html margins.

**Comment by dd32** [comment]
Semantically, I think `position: fixed` works better here anyway, as `position: sticky` is designed for elements that scroll into the viewport rather than being fixed to it permanently. Unless I'm missing something between Flexbox & sticky.

As noted above, it does mess with the style a bit.

With `.wp-block-group.global-header` defined with a width (100%) & height (62px..), and altering the admin bar rule to `html { margin-top: calc(var(--wp-global-header-height, 0px) + var(--wp-admin--admin-bar--height, 0px)); }`.. I get this:

https://user-images.githubusercontent.com/767313/152906488-d4e26c1f-3ff8-48fe-9f13-9e851267f73b.mov

Needs extra styles for mobile and other viewports, but swapping to fixed positioning does seem like a reasonable avenue of fixing this unless I missed something :)

**Comment by ryelle** [comment]
> for elements that scroll into the viewport rather than being fixed to it permanently.

I suppose that's a good point, I think `sticky` was used here since it keeps the space the element takes up in the document, but since we have the height custom properties, it should be fairly straightforward to add the margin offset like you're suggesting. 

I think since `fixed` takes this element out of the document flow, it avoids the underlying issue here.
